Build warning "'matchLevel' is deprecated"

This warning has been around for some while in both the package and the compiler builds. Would be good to clear this.

```console
Sources/_StringProcessing/ConsumerInterface.swift:139:12: warning build: 'matchLevel' is deprecated
```

--- a/Sources/_StringProcessing/MatchingOptions.swift
+++ b/Sources/_StringProcessing/MatchingOptions.swift
@@ -117,7 +117,6 @@ extension MatchingOptions {
 
 // Deprecated CharacterClass.MatchLevel API
 extension MatchingOptions {
-  @available(*, deprecated)
   var matchLevel: _CharacterClassModel.MatchLevel {
     switch semanticLevel {
     case .graphemeCluster:

